core/jobs/transforms/job_result_transforms.py

```diff
@@ -0,0 +1,149 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Provides an Apache Beam API for operating on NDB models."""
+
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from core.jobs.types import job_run_result
+
+import apache_beam as beam
+import result
+from typing import Any, Optional, Tuple
+
+
+class ResultsToJobRunResults(beam.PTransform):
+    """Transforms result.Result into job_run_result.JobRunResult."""
+
+    def __init__(
+        self, prefix: Optional[str] = None, label: Optional[str] = None
+    ) -> None:
+        """Initializes the ResultsToJobRunResults PTransform.
+
+        Args:
+            prefix: str|None. The prefix for the result string.
+            label: str|None. The label of the PTransform.
+        """
+        super().__init__(label=label)
+        self.prefix = '%s ' % prefix if prefix else ''
+
+    # This is needed because the Beam annotations validator doesn't properly
+    # work with result.Result.
+    @beam.typehints.no_annotations
+    def _transform_result_to_job_run_result(
+        self, result_item: result.Result[Any, Any]
+    ) -> job_run_result.JobRunResult:
+        """Transforms Result objects into JobRunResult objects. When the result
+        is Ok then transform it into stdout, if the result is Err transform it
+        into stderr.
+
+        Args:
+            result_item: Result. The result object.
+
+        Returns:
+            JobRunResult. The JobRunResult object.
+        """
+        if isinstance(result_item, result.Ok):
+            return job_run_result.JobRunResult.as_stdout(
+                '%sSUCCESS:' % self.prefix
+            )
+        else:
```

Comment:
Is it always going to be either Ok or Err? If not, then have an if / elif / else where the else throws an exception. If so, then link to a source that documents that Ok or Err are the only two choices.

Reply:
Yes this is always either `Ok` or `Err`, here is the documentation for it https://github.com/rustedpy/result, here is a link to the code https://github.com/rustedpy/result/blob/master/result/result.py#L236-L249

I'm not afraid this will ever change since the Result type is a common structure that is used in many different languages https://en.wikipedia.org/wiki/Result_type